tensorflow_quantum/core/ops/batch_util.py

```diff
@@ -538,30 +412,18 @@ def batch_sample(circuits, param_resolvers, n_samples, simulator):
 
     biggest_circuit = max(len(circuit.all_qubits()) for circuit in circuits)
     return_mem_shape = (len(circuits), n_samples, biggest_circuit)
-    shared_array = _make_simple_view(return_mem_shape, -2, np.int32, 'i')
-
-    if isinstance(simulator, cirq.DensityMatrixSimulator):
-        post_process = lambda state, size, n_samples: \
-            cirq.sample_density_matrix(
-                state.final_density_matrix, [i for i in range(size)],
-                repetitions=n_samples)
-    elif isinstance(simulator, cirq.Simulator):
-        post_process = lambda state, size, n_samples: cirq.sample_state_vector(
-            state.final_state_vector, list(range(size)), repetitions=n_samples)
-    else:
-        raise TypeError('Simulator {} is not supported by batch_sample.'.format(
-            type(simulator)))
-
-    input_args = list(
-        _prep_pool_input_args(range(len(circuits)), circuits, param_resolvers,
-                              [n_samples] * len(circuits)))
+    return_array = np.ones(return_mem_shape, dtype=np.int8) * -2
 
-    with ProcessPool(processes=None,
-                     initializer=_setup_dict,
-                     initargs=(shared_array, return_mem_shape, simulator,
-                               post_process)) as pool:
+    for batch, (c, resolver) in enumerate(zip(circuits, param_resolvers)):
+        if len(c.all_qubits()) == 0:
+            continue
 
-        pool.starmap(_sample_worker_func, input_args)
+        qb_keys = [(q, str(i)) for i, q in enumerate(sorted(c.all_qubits()))]
+        c_m = c + cirq.Circuit(cirq.measure(q, key=i) for q, i in qb_keys)
+        run_c = cirq.resolve_parameters(c_m, resolver)
+        bits = simulator.sample(run_c, repetitions=n_samples)
+        flat_m = bits[[x[1] for x in qb_keys]].to_numpy().astype(np.int8)
+        return_array[batch, :, :len(qb_keys)] = flat_m[:, ::-1]
```

Comment:
Would it work to do one fewer cycles of reversing: `return_array[batch, :, :len(qb_keys)] = flat_m[:, ::-1]` -> `return_array[batch, :, biggest_circuit - len(qb_keys):] = flat_m`, then just `return return_array`?

Reply:
Oops, good call. Done.